core/src/main/java/com/linecorp/armeria/client/metric/DefaultClientMeterIdPrefixFunction.java

```diff
@@ -0,0 +1,132 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.metric;
+
+import static com.google.common.base.MoreObjects.firstNonNull;
+import static com.linecorp.armeria.internal.common.metric.DefaultMeterIdPrefixFunction.addHttpStatus;
+import static java.util.Objects.requireNonNull;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.base.MoreObjects;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableList.Builder;
+
+import com.linecorp.armeria.client.ClientRequestContext;
+import com.linecorp.armeria.client.Endpoint;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.common.logging.RequestOnlyLog;
+import com.linecorp.armeria.common.metric.MeterIdPrefix;
+import com.linecorp.armeria.internal.common.util.TemporaryThreadLocals;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+
+final class DefaultClientMeterIdPrefixFunction implements ClientMeterIdPrefixFunction {
+
+    private static final Logger logger = LoggerFactory.getLogger(DefaultClientMeterIdPrefixFunction.class);
+    private static boolean warnedServiceRequestContext;
+
+    private final String name;
+    private final boolean includeHttpStatus;
+    private final boolean includeMethod;
+    private final boolean includeRemoteAddress;
+    private final boolean includeService;
+
+    DefaultClientMeterIdPrefixFunction(String name, boolean includeHttpStatus, boolean includeMethod,
+                                       boolean includeRemoteAddress, boolean includeService) {
+        this.name = requireNonNull(name, "name");
+        this.includeHttpStatus = includeHttpStatus;
+        this.includeMethod = includeMethod;
+        this.includeRemoteAddress = includeRemoteAddress;
+        this.includeService = includeService;
+    }
+
+    @Override
+    public MeterIdPrefix activeRequestPrefix(MeterRegistry registry, RequestOnlyLog log) {
+        /* method, remoteAddress, service */
+        final Builder<Tag> tagListBuilder = ImmutableList.builderWithExpectedSize(3);
+        addActiveRequestTags(tagListBuilder, log);
+        return new MeterIdPrefix(name, tagListBuilder.build());
+    }
+
+    private void addActiveRequestTags(Builder<Tag> tagListBuilder, RequestOnlyLog log) {
+        // Add tags in lexicographical order of the tag key.
+        if (includeMethod) {
+            tagListBuilder.add(Tag.of("method", log.name()));
+        }
+        if (includeRemoteAddress) {
+            tagListBuilder.add(Tag.of("remoteAddress", remoteAddress(log)));
+        }
+        if (includeService) {
+            tagListBuilder.add(Tag.of("service", firstNonNull(log.serviceName(), "none")));
+        }
+    }
+
+    private static String remoteAddress(RequestOnlyLog log) {
```

Comment:
Question) if `ctx.remoteAddress` exists, should this be considered first?

e.g. if an endpoint doesn't have a resolved ip address, the remote ip address will never be recorded

Reply:
Yeah, we can use that. Somehow, I thought we couldn't use it. 😓 
Fixed. 😉